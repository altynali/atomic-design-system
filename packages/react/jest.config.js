/** @type {import('jest').Config} */

const config = {
  verbose: true,
  testEnvironment: "jsdom",
  collectCoverageFrom: ["src/**/*.{js,ts, tsx}"],
  roots: ["<rootDir>/src"],
}

module.exports = config
