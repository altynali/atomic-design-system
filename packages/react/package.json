{
  "name": "atomic-design-system-react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "devDependencies": {
    "@types/node": "^20.10.0",
    "@types/react": "^18.2.23",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^4.6.0",
    "rollup-plugin-typescript2": "^0.36.0",
    "typescript": "^5.3.2"
  },
  "dependencies": {},
  "scripts": {
    "build": "rollup -c --bundleConfigAsCjs",
    "dev": "yarn build --watch"
  }
}
