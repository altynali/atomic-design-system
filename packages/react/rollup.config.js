import Ts from "rollup-plugin-typescript2"

const config = {
  input: [
    "src/index.ts",
    "src/atoms/Color/index.ts",
    "src/atoms/Text/index.ts",
    "src/atoms/Margin/index.ts",
    "src/atoms/Button/index.ts",
    "src/atoms/Padding/index.ts",
    "src/atoms/Grid/index.ts",
    "src/atoms/Checkbox/index.ts",
    "src/molecules/Card/index.ts",
    "src/molecules/Select/index.ts",
    "src/molecules/Input/index.ts",
    "src/molecules/CardList/index.ts",
    "src/organisms/Form/index.ts",
    "src/templates/Layout/index.ts",
  ],
  output: [
    {
      dir: "lib",
      format: "esm",
      sourcemap: true,
      preserveModules: true,
    },
  ],
  plugins: [Ts()],
  external: ["react", "atomic-design-system-foundation"],
}

export default config
