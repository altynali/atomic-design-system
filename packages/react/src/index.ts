import { Color } from "./atoms/Color"
import { Text } from "./atoms/Text"
import { Margin } from "./atoms/Margin"
import { Button } from "./atoms/Button"
import { Select } from "./molecules/Select"

export { Color, Text, Margin, Button, Select }

// TODO: import * from './atoms';
