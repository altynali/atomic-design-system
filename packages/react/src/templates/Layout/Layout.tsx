import React, { FC, PropsWithChildren } from "react"
import { Header } from "../../molecules/Header"
import { Footer } from "../../molecules/Footer"
import { classNames } from "../../utils"

export type LayoutProps = {
  className?: string
  noHeader?: boolean
  noFooter?: boolean
  label?: string
} & PropsWithChildren

const Layout: FC<LayoutProps> = (props) => {
  const {
    children,
    noHeader = false,
    noFooter = false,
    label = "Todo App",
    className = "",
  } = props

  return (
    <div className={classNames(className, ["atds-layout"])}>
      {!noHeader && <Header leftText="Logo" rightText="Header" />}
      <div className="atds-layout__content">
        <main>
          <h1>{label}</h1>
          {children}
        </main>
      </div>
      {!noFooter && <Footer leftText="Footer" rightText="2024" />}
    </div>
  )
}

export default Layout
