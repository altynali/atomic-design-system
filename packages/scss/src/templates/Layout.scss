@import "foundation/all";

.atds-layout {
  height: 100%;
  display: flex;
  min-height: 100vh;
  flex-direction: column;
  justify-content: space-between;
  box-sizing: border-box;
  &__content {
    @include padding("md");
  }
}
