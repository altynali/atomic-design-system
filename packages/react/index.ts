import { Color, Margin, Select, Text } from "./src/index"

export { Color, Text, Margin, Select }

// TODO: import * from './atoms';
