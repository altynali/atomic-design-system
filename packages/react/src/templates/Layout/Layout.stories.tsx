import React from "react"
import Layout from "./Layout"
import "atomic-design-system-scss/lib/Form.css"
import "atomic-design-system-scss/lib/Button.css"
import "atomic-design-system-scss/lib/Input.css"
import "atomic-design-system-scss/lib/Select.css"
import "atomic-design-system-scss/lib/Margin.css"
import "atomic-design-system-scss/lib/Padding.css"
import "atomic-design-system-scss/lib/Layout.css"
import { Form } from "../../organisms/Form"
import { Input } from "../../molecules/Input"
import { Select } from "../../molecules/Select"
import { Padding } from "../../atoms/Padding"
import { Spacing } from "atomic-design-system-foundation/lib/src"

export default {
  title: "Templates|Layout",
}

const options = [
  {
    label: "Strict Black",
    value: "black",
  },
  {
    label: "Heavenly Green",
    value: "green",
  },
  {
    label: "Sweet Pink",
    value: "pink",
  },
]

export const Common = () => (
  <Layout>
    <Form label="Form Label">
      <Padding bottom space={Spacing.sm}>
        <Input />
      </Padding>
      <Padding bottom space={Spacing.sm}>
        <Select options={options} />
      </Padding>
    </Form>
  </Layout>
)
