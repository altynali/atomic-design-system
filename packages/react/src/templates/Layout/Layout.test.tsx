import React from "react"
import { render } from "@testing-library/react"
import Layout from "./Layout"
import "@testing-library/jest-dom"

describe("Layout component", () => {
  it("renders layout with default label and children", () => {
    const { getByText } = render(
      <Layout>
        <div>Content</div>
      </Layout>
    )
    const label = getByText("Todo App")
    const content = getByText("Content")
    expect(label).toBeInTheDocument()
    expect(content).toBeInTheDocument()
  })

  it("renders layout without header if noHeader prop is true", () => {
    const { queryByTestId } = render(
      <Layout noHeader>
        <div>Content</div>
      </Layout>
    )
    const header = queryByTestId("header")
    expect(header).toBeNull()
  })

  it("renders layout without footer if noFooter prop is true", () => {
    const { queryByTestId } = render(
      <Layout noFooter>
        <div>Content</div>
      </Layout>
    )
    const footer = queryByTestId("footer")
    expect(footer).toBeNull()
  })

  it("applies custom className", () => {
    const { container } = render(
      <Layout className="custom">
        <div>Content</div>
      </Layout>
    )
    const layout = container.querySelector(".atds-layout")
    expect(layout).toHaveClass("custom")
  })

  it("renders layout with custom label", () => {
    const { getByText } = render(
      <Layout label="Custom Label">
        <div>Content</div>
      </Layout>
    )
    const label = getByText("Custom Label")
    expect(label).toBeInTheDocument()
  })

  test("snapshot of the base state", () => {
    const { asFragment } = render(<Layout />)

    expect(asFragment()).toMatchSnapshot()
  })
})
