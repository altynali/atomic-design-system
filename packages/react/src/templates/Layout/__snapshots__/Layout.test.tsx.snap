// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component snapshot of the base state 1`] = `
<DocumentFragment>
  <div
    class=" atds-layout"
  >
    <header
      class=" atds-footer"
    >
      <p
        class="undefined atds-text atds-text-base"
      >
        Logo
      </p>
      <p
        class="undefined atds-text atds-text-base"
      >
        Header
      </p>
    </header>
    <div
      class="atds-layout__content"
    >
      <main>
        <h1>
          Todo App
        </h1>
      </main>
    </div>
    <footer
      class=" atds-footer"
    >
      <p
        class="undefined atds-text atds-text-base"
      >
        Footer
      </p>
      <p
        class="undefined atds-text atds-text-base"
      >
        2024
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
